src/main/scala/gitbucket/core/service/HandleCommentService.scala

```diff
@@ -29,25 +36,31 @@ trait HandleCommentService {
         case (owner, name) =>
           val userName = loginAccount.userName
 
-          val (action, actionActivity) = actionOpt
+          actionOpt.collect {
+            case "close" if !issue.closed =>
+              updateClosed(owner, name, issue.issueId, true)
+            case "reopen" if issue.closed =>
+              updateClosed(owner, name, issue.issueId, false)
+          }
+
+          val (action, _) = actionOpt
```

Comment:
I could definitely simplify this further, but a combination of not being too familiar with this specific part, and being tired from this refactor is making me put this off until later.

Reply:
Okay, I got it. 😅